### ISSUE TITLE
[Question] How to backup

### ISSUE THREAD

**Issue by airon90** [yiotro/Antiyoy#28]
I need to format my smartphone but I'd like to save the data. How could I do that?

**Comment by yiotro** [comment]
Well, if you're advanced user then you can use something like titanium backup, but I personally would be too lazy to mess with that.

Currently there is no other way to backup your progress. But there is a way to skip to any level, if you're interested then email me and I'll tell you how to do it :)
My email: yiotro93@gmail.com

**Comment by airon90** [comment]
I didn't root my phone so I can't use Titanium Backup. 
Let's think about a backup system (import/export data). By now, I think that the only data to be saved are:
- Progress in campain
- Saves
- Replays
- Settings

I keep the bug open, so that if somebody wants can implement it. If owner doesn't want such a system, close this bug ;)

I don't think I will contact you, otherwise I will always use the cheating system :D

**Comment by yiotro** [comment]
You forgot about regular saves :)

**Comment by abstractionmage** [comment]
Hi, do you think there is a way to access my Anityoy campaign progress somehow? Is there like a hidden folder or something I didn't see? 
Because I'd like to set up an automatic cloud backup just in case though I couldn't find any of the game files, the "Redmi Note 4\Internal shared storage\Android\data\yio.tro.antiyoy.android/files" folder is completely empty for me

**Comment by yiotro** [comment]
I've heard that some people were able to transfer progress. All save data is stored using android preferences. There should be 'xml' files somewhere (but I don't know details).

**Comment by MoonAndTheMoon** [comment]
When I look at it with a file manager for a rooted emulator the location looks like this:
![bilde](https://user-images.githubusercontent.com/35038532/72227303-2f6b8500-359b-11ea-8a81-8b01772b1f0c.png)
(_cache_ and _files_ folders are found alongside _shared_prefs_)

But access to this area is pretty restricted though I don't know the specifics around it. In other words "..you probably won't find it".

I know another open source game that stores data in its own folder on the internal storage which makes backups easy. I hypothesize this data won't be deleted when the app is uninstalled, and perhaps it would be not-so-cool if every app did this.

But perhaps an option to export/import game data to/from internal storage would be possible? The folder name can be static so there's no need for find folder dialogs.

**Comment by vertigo220** [comment]
This really should be added. I love this game and have spent many, many hours playing it, all the way through the campaign and most of the user levels. While playing through all that again wouldn't necessarily be the worst thing, it's not really something I _want_ to do, especially just to lose all my progress again the next time I have to switch devices. This is, IMO, one of the biggest issues with Android in general and one of the reasons I've lost ambition to even do much with it anymore (that and apps constantly breaking due to Google's changes, causing apps I've come to love and rely on to become useless).

I've always appreciated that yiotro's games require zero permissions, and I suspect storage permissions would be necessary to do this properly, but I see no reason this couldn't be done some other way, at least to some extent. Even just providing a screen to display and edit the various numbers, i.e. last campaign level cleared, names (or numbers) of user levels cleared, and stats. Sure, this would allow people to cheat, but if they really want to do that, so what? They're only cheating themselves, anyway. At least allow those of us that have sunk countless hours into the game to not lose all that progress every few years, give or take.

Another possible option would be to have the game create a checksum based on all this information, which could then be entered into it on another device, which would use it to determine all the values. The point is there are ways to do this while avoiding storage permissions, assuming that's the goal. Or just add the permissions, since it won't have any others, so they wouldn't be a "risk." I've been wanting to play this again lately, but can't bring myself to start back from the very beginning.

**Comment by yiotro** [comment]
I agree, game would be better with some kind of a way to sync your progress between devices. 

In my recent games I actually do just that. For example, in S-kladom there is a small 'code' that consists of five icons. To restore your progress on a new device you just enter this code. 

In antiyoy there is an easy way to just continue from where you stopped on previous device. Watch this short video: https://youtube.com/shorts/FMYOrE5VaJ4

It shows how to open a secret screen. In that screen then just tap 'unlock levels', now you can launch any campaign level.

**Comment by vertigo220** [comment]
Yeah, I saw the secret page, but that has to be done for each individual level, the level presumably has to be beat to keep it unlocked ("unlocked" one multiple times but when I went back to the menu it was locked), and it doesn't work for user levels or restore saved games and stats.

Why not add something like the S-kladom icon code to this?

**Comment by yiotro** [comment]
I don't remember why I didn't do that. I'll think about adding such option.

**Comment by vertigo220** [comment]
Thanks. That would be amazing. I'm keeping it on my old phone just in case, so if you add it I can update it on that phone and use the new feature to restore my progress on my new one.
